src/vs/platform/storage/common/profileStorageService.ts

```diff
@@ -0,0 +1,151 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Event } from 'vs/base/common/event';
+import { Disposable, isDisposable } from 'vs/base/common/lifecycle';
+import { IStorage, IStorageDatabase, Storage } from 'vs/base/parts/storage/common/storage';
+import { createDecorator } from 'vs/platform/instantiation/common/instantiation';
+import { AbstractStorageService, IStorageService, IStorageValueChangeEvent, StorageScope, StorageTarget } from 'vs/platform/storage/common/storage';
+import { IUserDataProfile } from 'vs/platform/userDataProfile/common/userDataProfile';
+
+export interface IProfileStorageValueChanges {
+	readonly profile: IUserDataProfile;
+	readonly changes: IStorageValueChangeEvent[];
+}
+
+export interface IProfileStorageChanges {
+	readonly targetChanges: IUserDataProfile[];
+	readonly valueChanges: IProfileStorageValueChanges[];
+}
+
+export interface IStorageValue {
+	readonly value: string | undefined;
+	readonly target: StorageTarget;
+}
+
+export const IProfileStorageService = createDecorator<IProfileStorageService>('IProfileStorageService');
+export interface IProfileStorageService {
```

Comment:
It would help (me) to explain the intent of the profile storage service. I wonder if the name is also a bit misleading because the methods are not coming close the `IStorageService`, and we would want to avoid that someone is accidentally using this thing?

Reply:
This service allows consumers to read and write into any profile storage - given a profile. It also emits an event when any profile storage changes.

This is the requirement as it is for syncing global state - default and for all profiles.